services/slam/builtin/builtin.go

```diff
@@ -815,14 +815,16 @@ func (slamSvc *builtIn) getAndSaveDataSparse(
 		}
 
 		image, release, err := slamSvc.getPNGImage(ctx, cams[0])
+		if release != nil {
```

Comment:
how did you test this was the issue?

Reply:
I ran tests of 10,000 trials: `go test . -run TestOrbslamYAMLNew -race -count 10000`

Reply:
coo